libs/features/personalization/personalization.js

```diff
@@ -216,11 +216,67 @@ function normalizeKeys(obj) {
   }, {});
 }
 
+const getDivInTargetedCell = (tableCell) => {
+  tableCell.replaceChildren();
+  const div = document.createElement('div');
+  tableCell.appendChild(div);
+  return div;
+};
+
+const querySelector = (el, selector, all = false) => {
+  try {
+    if (all) {
+      return el.querySelectorAll(selector);
+    }
+    return el.querySelector(selector);
+  } catch (e) {
+    /* eslint-disable-next-line no-console */
+    console.log('Invalid selector: ', selector);
+    return null;
+  }
+};
+
+function getSelectedEl(rootEl, selector) {
+  if (!selector) return null;
+
+  let selectedEl = querySelector(rootEl, selector);
```

Comment:
shouldn't we rather do before this string tests on selector for `<blockname>-<num>` or cell regexp? i guess those two cases are supposed to be seldom, but we never know where that goes, and i guess string tests should be quicker than empty DOM query.